app/pulp/app/serializers/base.py

```diff
@@ -0,0 +1,173 @@
+        return super(GenericKeyValueRelatedField, self).to_representation(value.mapping)
+
+
+class ModelSerializer(serializers.HyperlinkedModelSerializer):
+    """Base serializer for use with :class:`pulp.app.models.Model`
+
+    This ensures that all Serializers provide values for the '_href` field, and
+    adds read/write support for :class:`pulp.app.serializers.GenericKeyValueRelatedField`
+    nested fields.
+    """
+
+    class Meta:
+        fields = ('_href',)
+
+    def create(self, validated_data):
+        # pop related fields out of validated data
+        generic_field_mappings = self._generic_field_mappings(validated_data)
+
+        instance = super(ModelSerializer, self).create(validated_data)
+
+        # populate related fields
+        self._populate_generic_fields(instance, generic_field_mappings)
+
+        return instance
+
+    def update(self, instance, validated_data):
+        # pop related fields out of validated data
+        generic_field_mappings = self._generic_field_mappings(validated_data)
+
+        instance = super(ModelSerializer, self).update(instance, validated_data)
+
+        # populate related fields
+        self._populate_generic_fields(instance, generic_field_mappings)
+
+        return instance
+
+    def _generic_field_mappings(self, validated_data):
+        # Strip generic k/v pairs out of validated data and return them.
+        generic_mappings = {}
+        for field_name, field in self.get_fields().items():
+            if isinstance(field, GenericKeyValueRelatedField):
+                try:
+                    generic_mappings[field_name] = validated_data.pop(field_name)
+                except KeyError:
+                    pass
+        return generic_mappings
+
+    def _populate_generic_fields(self, instance, field_mappings):
+        for field_name, mapping in field_mappings.items():
+            field = getattr(instance, field_name)
+            field.mapping.replace(mapping)
+
+
+class MasterModelSerializer(ModelSerializer):
+    """
+    Base serializer for all Master/Detail Models.
+
+    When subclassing this, all subclasses should explicitly inherit the fields of their parent
+    in their Meta options class. For example:
+
+        class MasterSerializer(MasterModelSerializer):
+            foo = SerializerField()
+
+            class Meta:
+                fields = MasterModelSerializer.Meta.fields + ('foo',)
+
+        class DetailSerializer(MasterSerializer):
+            bar = SerializerField()
+
+            class Meta:
+                fields = MasterSerializer.Meta.fields + ('bar',)
+
+    This ensures that fields are represented consistently throughout the API.
+
+    Other Meta attributes, such as `filter_fields`, should also be inherited in this way
+    as-needed.
+
+    """
+    type = serializers.CharField(read_only=True)
+
+    class Meta:
+        fields = ModelSerializer.Meta.fields + ('type',)
+
+
+class DetailRelatedField(serializers.HyperlinkedRelatedField):
```

Comment:
In tinkering with this a little bit, the need for a very similar `DetailIdentityField` has become apparent.